src/storage/v2/storage.cpp

```diff
@@ -1718,11 +1756,37 @@ void Storage::CollectGarbage() {
       MG_ASSERT(edge_acc.remove(edge), "Invalid database state!");
     }
   }
+
+  // EXPENSIVE full scan, is only run if an IN_MEMORY_ANALYTICAL transaction involved any deletions
```

Comment:
Did you benchmark how much time it takes to do a full scan? I have really no perspective on this how much time it takes

Reply:
No, but it is a relative behaviour.
Unlike in transaction mode, where we have a precise collection of what transactions have completed + the deltas made on each of those transactions. A situations like 1'000'000's of vertices with only one deletion is fast.
Yet in analytical, it will require a full scan, 1'000'000's of vertices with only one deletion is slow compared to the transaction mode.

Quick testing, `UNWIND RANGE(1, 1000000) AS i CREATE ();` then `MERGE (n) DELETE n; FREE MEMORY;` under the two storage modes.

IN_MEMORY_TRANSACTIONAL : round trip in 0.010 sec
IN_MEMORY_ANALYTICAL : round trip in 0.560 sec

Reply:
Okay, this seems fine. If we people will complain we can iterate.